bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/protocol/NikobusConnection.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nikobus.internal.protocol;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.TooManyListenersException;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEvent;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEventListener;
+import org.eclipse.smarthome.io.transport.serial.SerialPortIdentifier;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NikobusConnection } is responsible for creating connections to clients.
+ *
+ * @author Boris Krivonog - Initial contribution
+ */
+public class NikobusConnection implements SerialPortEventListener {
+    private final Logger logger = LoggerFactory.getLogger(NikobusConnection.class);
+    private final SerialPortManager serialPortManager;
+    private final String portName;
+    private final Consumer<Byte> processData;
+    private @Nullable SerialPort serialPort;
+
+    public NikobusConnection(SerialPortManager serialPortManager, String portName, Consumer<Byte> processData) {
+        this.serialPortManager = serialPortManager;
+        this.portName = portName;
+        this.processData = processData;
+    }
+
+    /**
+     * Return true if this manager is connected.
+     *
+     * @return
+     */
+    public boolean isConnected() {
+        return serialPort != null;
+    }
+
+    /**
+     * Connect to the receiver.
+     *
+     **/
+    public void connect() throws IOException {
+        if (isConnected()) {
+            return;
+        }
+
+        SerialPortIdentifier portId = serialPortManager.getIdentifier(portName);
+        if (portId == null) {
+            throw new IOException("Port is not known!");
```

Comment:
Maybe include the port name:
```suggestion
            throw new IOException("Port '" + portName + "' is not known!");
```